### PR TITLE
Use WebView in the LinkPreviewDialog

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T314003

### CODE REVIEW

File: app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt [hunk 1]
```diff
@@ -248,9 +251,14 @@ class LinkPreviewDialog : ExtendedBottomSheetDialogFragment(), LinkPreviewErrorV
...
     }
 
+    @SuppressLint("RequiresFeature")
     private fun setPreviewContents(contents: LinkPreviewContents) {
         if (!contents.extract.isNullOrEmpty()) {
-            binding.linkPreviewExtract.text = StringUtil.fromHtml(contents.extract)
+            binding.linkPreviewExtractWebview.setBackgroundColor(Color.TRANSPARENT)
+            binding.linkPreviewExtractWebview.setLayerType(WebView.LAYER_TYPE_SOFTWARE, null)
+            // Color hex code from android:textColorPrimary
+            val colorHex = if (WikipediaApp.instance.currentTheme.isDark) "#f8f9fa" else "#202122"
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt:260]: These values should not be hard-coded here, and can be computed from the actual color resource.

File: app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt [hunk 2]
```diff
@@ -248,9 +251,14 @@ class LinkPreviewDialog : ExtendedBottomSheetDialogFragment(), LinkPreviewErrorV
         }
     }
 
+    @SuppressLint("RequiresFeature")
     private fun setPreviewContents(contents: LinkPreviewContents) {
         if (!contents.extract.isNullOrEmpty()) {
-            binding.linkPreviewExtract.text = StringUtil.fromHtml(contents.extract)
+            binding.linkPreviewExtractWebview.setBackgroundColor(Color.TRANSPARENT)
+            binding.linkPreviewExtractWebview.setLayerType(WebView.LAYER_TYPE_SOFTWARE, null)
```
[the remark this answers is not in the document]
Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt]: Removed!

File: app/build.gradle [hunk 3]
```diff
@@ -163,6 +163,8 @@ apply from: '../gradle/src/checkstyle.gradle'
 apply from: '../gradle/src/ktlint.gradle'
 
 dependencies {
+    implementation 'androidx.webkit:webkit:1.4.0'
```
[the remark this answers is not in the document]
Comment (cooltey) [reply app/build.gradle]: No. Removed.

File: app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt]: I tried to use `Integer.toHexString()` but the regular 6 digits hex string will have an additional `ff` at the beginning of the text after converting it. I use `substring()` to start the index from the 3rd text in order to display the color correctly. Please let me know if you have a better solution for that, thanks!

**Comment by dbrant** [comment]
Merging, with a heavy heart.
Let's keep a close eye on ANRs and crashes after this is released.

**Pull request merged**